src/lib/systemd.ts

```diff
@@ -1,113 +1,130 @@
 import * as Bluebird from 'bluebird';
-import * as dbus from 'dbus-native';
+import * as dbus from 'dbus';
+import TypedError = require('typed-error');
 
-const bus = dbus.systemBus();
-const invokeAsync = Bluebird.promisify(bus.invoke).bind(bus);
+export class DbusError extends TypedError {}
+
+const bus = dbus.getBus('system');
+const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
+	context: bus,
+});
 
 const defaultPathInfo = {
 	path: '/org/freedesktop/systemd1',
 	destination: 'org.freedesktop.systemd1',
 	interface: 'org.freedesktop.systemd1.Manager',
 };
 
-const systemdManagerMethodCall = (
-	method: string,
-	signature?: string,
-	body?: any[],
+async function getInterface(
 	pathInfo: {
 		path: string;
 		destination: string;
 		interface: string;
 	} = defaultPathInfo,
-) => {
-	if (signature == null) {
-		signature = '';
-	}
-	if (body == null) {
-		body = [];
+) {
+	try {
+		return await getInterfaceAsync(
+			pathInfo.destination,
+			pathInfo.path,
+			pathInfo.interface,
+		);
+	} catch (e) {
+		throw new DbusError(e);
 	}
-	return invokeAsync({
-		...pathInfo,
-		member: method,
-		signature,
-		body,
-	});
-};
+}
 
-export function restartService(serviceName: string) {
-	return systemdManagerMethodCall('RestartUnit', 'ss', [
-		`${serviceName}.service`,
-		'fail',
-	]);
+export async function restartService(serviceName: string) {
+	try {
+		const systemd = await getInterface();
```

Comment:
This `getInterface` call should be moved outside of the `try`, otherwise it can throw a `new DbusError(new DbusError(e))`.